src/porepy/numerics/ad/operators.py

```diff
@@ -1234,15 +1132,232 @@ def _parse_other(self, other):
             return [self, DenseArray(other)]
         elif isinstance(other, sps.spmatrix):
             return [self, SparseArray(other)]
-        elif isinstance(other, Operator):
-            return [self, other]
         elif isinstance(other, AdArray):
             # This may happen when using nested pp.ad.Function.
             return [self, other]
+        elif isinstance(other, pp.ad.SecondaryOperator):
+            # Putting secondary operator above abstract function, because it inherits
+            # from abstract function, but is operable
+            return [self, other]
+        elif isinstance(other, pp.ad.AbstractFunction):
+            # Need to put this here, because overload of AbstractFunction is not
+            # applied if AbstractFunction is right operand.
+            raise TypeError(
+                "Operator functions must be called before applying any operation."
+            )
+        elif isinstance(other, Operator):
+            # Put Operator at end, because Seconary and Abstract are also operators
+            return [self, other]
         else:
             raise ValueError(f"Cannot parse {other} as an AD operator")
 
 
+class TimeDependentOperator(Operator):
```

Comment:
New intermediate base class for operators with time-dependent behavior.
Implements the pushing back of the time step index in a general manner.